index.go

```diff
@@ -73,6 +73,8 @@ type IndexPart interface {
 	// Transform is a function that alters the value to be indexed as well as any search criteria.
 	// For example LowerCase is a Transform function that transforms the value to lower case.
 	Transform(value interface{}) interface{}
+	// Transformer return the Transform function
+	Transformer() Transform
```

Comment:
This feels really odd, having dependency which is both invoked (using `Transform`) but can be returned as well (`Transformer`)?

Reply:
changed fn definition to use interface{}